File: src/ui/src/models/api/Prompt/GetPromptsModel.ts
import { GetPromptsItemModel } from './GetPromptsItemModel';

/**
 * Model for getting prompts from the API
 */
export interface GetPromptsModel {
  /**
   * The total count of prompts
   */
  count: number;

  /**
   * The list of prompts
   */
  prompts: GetPromptsItemModel[];
  
  /**
   * The list of prompts (alternative field name for compatibility)
   */
  items?: GetPromptsItemModel[];
}

File: src/backend/DonkeyWork.Chat.Api/Models/Prompt/GetPromptsModel.cs
// ------------------------------------------------------
// <copyright file="GetPromptsModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Api.Models.Prompt;

/// <summary>
/// Gets the conversations model.
/// </summary>
public record GetPromptsModel
{
    /// <summary>
    /// Gets the count of conversations.
    /// </summary>
    public int Count { get; init; }

    /// <summary>
    /// Gets the conversations.
    /// </summary>
    public IEnumerable<GetPromptsItemModel> Prompts { get; init; } = [];
}